recipes/openssl/3.x.x/conanfile.py

```diff
@@ -0,0 +1,630 @@
+            "*BSD-armv8-*": "BSD-generic64",
+            "*BSD-mips64-*": "BSD-generic64",
+            "*BSD-ppc64-*": "BSD-generic64",
+            "*BSD-ppc64le-*": "BSD-generic64",
+            "*BSD-ppc64be-*": "BSD-generic64",
+            "AIX-ppc32-gcc": "aix-gcc",
+            "AIX-ppc64-gcc": "aix64-gcc",
+            "AIX-pcc32-*": "aix-cc",
+            "AIX-ppc64-*": "aix64-cc",
+            "AIX-*-*": "aix-common",
+            "*BSD-*-*": "BSD-generic32",
+            "Emscripten-*-*": "cc",
+            "Neutrino-*-*": "BASE_unix",
+        }
+
+    @property
+    def _ancestor_target(self):
+        if "CONAN_OPENSSL_CONFIGURATION" in os.environ:
+            return os.environ["CONAN_OPENSSL_CONFIGURATION"]
+        query = "%s-%s-%s" % (self.settings.os, self.settings.arch, self.settings.compiler)
+        ancestor = next((self._targets[i] for i in self._targets if fnmatch.fnmatch(query, i)), None)
+        if not ancestor:
+            raise ConanInvalidConfiguration("Unsupported configuration: %s %s %s, "
+                                            "please open an issue: "
+                                            "https://github.com/conan-io/conan-center-index/issues. "
+                                            "alternatively, set the CONAN_OPENSSL_CONFIGURATION environment variable "
+                                            "into your conan profile "
+                                            "(list of configurations can be found by running './Configure --help')." %
+                                            (self.settings.os,
+                                            self.settings.arch,
+                                            self.settings.compiler))
+        return ancestor
+
+    def _tool(self, env_name, apple_name):
+        if env_name in os.environ:
+            return os.environ[env_name]
+        if self.settings.compiler == "apple-clang":
+            return getattr(tools.XCRun(self.settings), apple_name)
+        return None
+
+    def _patch_configure(self):
+        # since _patch_makefile_org will replace binutils variables
+        # use a more restricted regular expresion to prevent that Configure script trying to do it again
+        configure = os.path.join(self._source_subfolder, "Configure")
+        tools.replace_in_file(configure, r"s/^AR=\s*ar/AR= $ar/;", r"s/^AR=\s*ar\b/AR= $ar/;")
+
+    def _patch_makefile_org(self):
+        # https://wiki.openssl.org/index.php/Compilation_and_Installation#Modifying_Build_Settings
+        # its often easier to modify Configure and Makefile.org rather than trying to add targets to the configure scripts
+        def adjust_path(path):
+            return path.replace("\\", "/") if tools.os_info.is_windows else path
+
+        makefile_org = os.path.join(self._source_subfolder, "Makefile.org")
+        env_build = self._get_env_build()
+        with tools.environment_append(env_build.vars):
+            if not "CROSS_COMPILE" in os.environ:
```

Comment:
The use of `CROSS_COMPILE` looks wrong to me.
I think it should test for `tools.cross_building(self.settings, skip_x86_64=True)`.

Reply:
I think this is a very ancient leftover which is worth to update in 1.x.x too

So let's spin this out in an own issue and then port the changes later